rustworkx-core/src/dag_algo.rs

```diff
@@ -599,3 +766,218 @@ mod test_lexicographical_topological_sort {
         assert_eq!(result, Ok(Some(vec![nodes[7]])));
     }
 }
+
+#[cfg(test)]
+mod test_collect_bicolor_runs {
```

Comment:
I tried my best migrating the python tests to rust but I skipped `test_color_with_ignored_edge` and `test_two_colors_with_barrier`.